lib/ui/pages/qrscanner_page.dart

```diff
@@ -0,0 +1,132 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:lottie/lottie.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:qrapp/services/qrservice.dart';
+import 'package:qrapp/ui/components/qrstatuswidget.dart';
+
+class QRScannerPage extends StatefulWidget {
+  const QRScannerPage({Key? key}) : super(key: key);
+
+  @override
+  _QRScannerPageState createState() => _QRScannerPageState();
+}
+
+class _QRScannerPageState extends State<QRScannerPage> {
+  final GlobalKey qrScannerKey = GlobalKey(debugLabel: 'QR Scanner');
+  late QRService provider;
+  Barcode? qrResults;
+  QRViewController? qrController;
+
+  bool scanningComplete = false;
+
+  @override
+  void initState() {
+    super.initState();
+    setState(() {
+      provider = Provider.of<QRService>(context, listen: false);
+    });
+  }
+
+  @override
+  void reassemble() {
+    super.reassemble();
+    if (Platform.isAndroid) {
+      qrController!.pauseCamera();
+    } else if (Platform.isIOS) {
+      qrController!.resumeCamera();
+    }
+  }
+
+  void _onScannerStart(QRViewController controller) async {
+    qrController = controller;
+    controller.scannedDataStream.listen((scanData) {
+      setState(() {
+        qrResults = scanData;
+        provider.verifySeed(context, scanData.code!);
+      });
+    });
+  }
+
+  @override
+  void dispose() {
+    if (qrController != null) {
+      qrController!.dispose();
+    }
+    super.dispose();
+  }
+
+  Widget buildScanningWindow(BuildContext context) {
+    if (context.watch<QRService>().isQRScanning == true) {
+      return Column(
```

Comment:
**suggestion:** even though this is internal to this widget, it's preferred to use classes instead of functions for widgets: https://stackoverflow.com/a/53234826

Reply:
I would also add that if you created a widget instead of a function, you could have made the widget `const` and use a `Consumer` to better control children rebuilds. This approach doesn't have a specific widget rebuild control logic which may be very inefficient on larger subtrees